clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -664,4 +666,55 @@ private static void writeLeaderChangeMessage(ByteBuffer buffer,
         builder.close();
     }
 
+    public static MemoryRecords withSnapshotHeaderRecord(
+            long initialOffset,
+            long timestamp,
+            int leaderEpoch,
+            ByteBuffer buffer,
+            MetadataSnapshotHeaderRecord snapshotHeaderRecord) {
+        writeSnapshotHeaderRecord(buffer, initialOffset, timestamp, leaderEpoch, snapshotHeaderRecord);
+        buffer.flip();
+        return MemoryRecords.readableRecords(buffer);
+            }
+
+    private static void writeSnapshotHeaderRecord(ByteBuffer buffer,
+            long initialOffset,
+            long timestamp,
+            int leaderEpoch,
+            MetadataSnapshotHeaderRecord snapshotHeaderRecord) {
+        MemoryRecordsBuilder builder = new MemoryRecordsBuilder(
```

Comment:
Does this need to be wrapped in a try... catch block to avoid leaking something if we throw an exception in `appendSnapshotHeaderMessage` ?

Reply:
You are right in that one of the underlying calls throws a few exceptions, but again looking at a similar implementation for the leader change message. Given that implementation (being canonical) does not implement a try/catch, I assumed that the API structure handles exceptions somewhere higher up in the stack. I guess that understanding is not correct. I will make similar changes for the leader change call stack as well then. Will ping offline about what the general exception handling mechanism here.